server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -243,7 +245,22 @@ public Sink getSink(long timestamp)
                       @Override
                       public QueryRunner<T> apply(TimelineObjectHolder<String, Sink> holder)
                       {
+                        if (holder == null) {
```

Comment:
is null here even possible, since we control what gets added to the querySinks list?